h5py/_hl/files.py

```diff
@@ -279,17 +279,18 @@ def swmr_mode(self, value):
             else:
                 raise ValueError("It is not possible to forcibly switch SWMR mode off.")
 
-    def __init__(self, name, mode=None, driver=None,
+    def __init__(self, file, mode=None, driver=None,
```

Comment:
I would avoid changing `name` to `file`, as there may be users using `name` via keyword arguments.

Reply:
Indeed, that might be undesirable.
[pyfive](https://github.com/jjhelmus/pyfive) uses "filename : str or file-like":
https://github.com/jjhelmus/pyfive/blob/46dc0bd19b9126d33e6b4eb7153451f3820103d3/pyfive/high_level.py#L152
Shall I just let `name` be a file-like object, or additionally introduce argument `file`, for cleanness?